pygeoapi/api/__init__.py

```diff
@@ -370,6 +375,16 @@ def with_data(cls, request, supported_locales) -> 'APIRequest':
                     api_req._data = loop.run_until_complete(request.body())
                 except ModuleNotFoundError:
                     LOGGER.error('Module nest-asyncio not found')
+            return api_req
+
+    @classmethod
+    def from_flask(cls, request: flask.Request, supported_locales
```

Comment:
We should adopt a `Protocol` for the incoming request from Web Frameworks and also adopt the dependency injection pattern to pass it to the `APIRequest ` class

Reply:
How would a `Protocol` look like for this? The different request classes have some different fields, so I'm not sure how it could be combined to one `Protocol`